src/expression/node/FunctionNode.js

```diff
@@ -101,34 +100,21 @@ export const createFunctionNode = /* #__PURE__ */ factory(name, dependencies, ({
         const rawArgs = this.args
         return function evalFunctionNode (scope, args, context) {
           const fn = resolveFn(scope)
-          return fn(rawArgs, math, Object.assign({}, scope, args))
+          return fn(rawArgs, math, createSubScope(scope, args), scope)
         }
       } else {
-        // "regular" evaluation
-        if (evalArgs.length === 1) {
-          const evalArg0 = evalArgs[0]
-          return function evalFunctionNode (scope, args, context) {
-            const fn = resolveFn(scope)
-            return fn(evalArg0(scope, args, context))
-          }
-        } else if (evalArgs.length === 2) {
-          const evalArg0 = evalArgs[0]
-          const evalArg1 = evalArgs[1]
-          return function evalFunctionNode (scope, args, context) {
-            const fn = resolveFn(scope)
-            return fn(evalArg0(scope, args, context), evalArg1(scope, args, context))
-          }
-        } else {
-          return function evalFunctionNode (scope, args, context) {
-            const fn = resolveFn(scope)
-            return fn.apply(null, map(evalArgs, function (evalArg) {
-              return evalArg(scope, args, context)
-            }))
-          }
+        return function evalFunctionNode (scope, args, context) {
```

Comment:
The three cases for 1, 2, or more than 2 arguments where here for performance: looping over a map is slower than just calling 1 or 2 functions. It would be nice to simplify the code, but please let's only do that after verifying that it doesn't have negative impact on the performance. It may be a good idea to add a short comment explaining this 😁

Reply:
Ok, I made a switch statement. How are you benchmarking/profiling to make these micro-optimizing decisions?